apps/multisig-signer/src/app/config/contracts/contracts-provider.tsx

```diff
@@ -0,0 +1,59 @@
+import { MultisigControl } from '@vegaprotocol/smart-contracts';
+import { Splash } from '@vegaprotocol/ui-toolkit';
+import React from 'react';
+import { ethers } from 'ethers';
+import type { ContractsContextShape } from './contracts-context';
+import { ContractsContext } from './contracts-context';
+import { useEthereumConfig } from '@vegaprotocol/web3';
+import { useEnvironment } from '@vegaprotocol/environment';
+
+/**
+ * Provides Vega Ethereum contract instances to its children.
+ */
+export const ContractsProvider = ({ children }: { children: JSX.Element }) => {
+  const { config } = useEthereumConfig();
+  const { VEGA_ENV, ETHEREUM_PROVIDER_URL } = useEnvironment();
+  const [contracts, setContracts] =
+    React.useState<ContractsContextShape | null>(null);
+
+  // Create instances of contract classes. If we have an account use a signer for the
+  // contracts so that we can sign transactions, otherwise use the provider for just
+  // reading data
+  React.useEffect(() => {
```

Comment:
chore (non blocking): for consistentcy import and use `useEffect` directly, same for other react imports in other files.